File: assign_test/data/room.js

// class Room {
//   id;
//   capacity;
//   type;
//   remaining;

//   constructor (roomDetails) {
//     this.id = roomDetails.id;
//     this.capacity = roomDetails.capacity;
//     this.type = roomDetails.type;
//     this.remaining = roomDetails.remaining;
//   }
// }

// export function saveToStorage() {
//   localStorage.setItem('rooms', JSON.stringify(rooms));
// }

// export let rooms = JSON.parse(localStorage.getItem('rooms')) || [];

// // Fetch dữ liệu từ file rooms.json và cập nhật vào biến rooms
// fetch('rooms.json')
//   .then(response => response.json())  // Chuyển dữ liệu JSON thành đối tượng JavaScript
//   .then(data => {
//     rooms = data.map(roomDetails => new Room(roomDetails));  // Tạo các đối tượng Room từ dữ liệu JSON

//     // Lưu vào localStorage để lần sau có thể lấy ra
//     localStorage.setItem('rooms', JSON.stringify(rooms));
//   })
//   .catch(error => console.error('Error fetching room data:', error));
// rooms = rooms.map((roomDetails) => {
//   return new Room(roomDetails);
// });

let data = `
[
  (1, 6, "NORMAL", 6),
  (2, 4, "DELUXE", 2),
  (3, 6, "NORMAL", 1),
  (4, 1, "VIP", 1),
  (5, 4, "DELUXE", 0),
  (6, 1, "VIP", 1),
  (7, 4, "DELUXE", 2),
  (8, 4, "DELUXE", 1)
]
`;

// Chuyển đổi chuỗi thành định dạng JSON
let jsonData = data
  .replace(/\(/g, '[') // Thay thế ( bằng [
  .replace(/\)/g, ']') // Thay thế ) bằng ]
  .replace(/,\s*\]/g, ']') // Xóa dấu phẩy trước dấu ]
  .replace(/"\s*\]/g, '"'); // Xóa dấu cách trước dấu ]

export let rooms = JSON.parse(jsonData);

File: assign_test/data/room.py
import json

# Giả sử rooms là danh sách các phòng
rooms = [{
  "id": 1,
  "capacity": 6,
  "type": "Normal",
  "remaining": 5
}, {
  "id": 2,
  "capacity": 6,
  "type": "Normal",
  "remaining": 4
}, {
  "id": 3,
  "capacity": 4,
  "type": "Deluxe",
  "remaining": 4
}, {
  "id": 4,
  "capacity": 1,
  "type": "VIP",
  "remaining": 1
}, {
  "id": 5,
  "capacity": 1,
  "type": "VIP",
  "remaining": 1
}, {
  "id": 6,
  "capacity": 4,
  "type": "Deluxe",
  "remaining": 0
}, {
  "id": 7,
  "capacity": 4,
  "type": "Deluxe",
  "remaining": 1
}, {
  "id": 8,
  "capacity": 4,
  "type": "Deluxe",
  "remaining": 1
}]

# Ghi dữ liệu rooms ra file JSON
with open('rooms.json', 'w') as json_file:
    json.dump(rooms, json_file)